### PR TITLE
Platform 63

### PR DESCRIPTION
**Pull request by jamccomb92**
### Description of proposed changes
_What does this pull request accomplish? Why should it be integrated?_
This adds the ability to check online status in the background of the patiendID form and then submit any PatientID forms that were submitted offline and saved to Async Storage. This also adds a GetAllData into the Async Storage that will retrieve all data stored in async storage (lol probably self explanatory)

### Checklist
_Put an `x` in the boxes that apply. _
- [x] My lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my changes work (if applicable)
- [ ] I have added or updated necessary documentation (if appropriate)

### Additional comments
Is there anything else you want us to know?

### CODE REVIEW

**Comment by jamccomb92** [comment]
@hopetambala It does appear that it may take longer than 1 go around (10 seconds) before it will submit every time. I am not clear as to why it will sometimes but not others, we may just want to space that out a bit more.

File: domains/DataCollection/Forms/PatientID/index.js [hunk 1]
```diff
@@ -23,10 +27,52 @@ import configArray from './config';
 // });
 
 const PatientIDForm = ({ navigation }) => {
+  // similar to componentDidMount and componenetWillUnmount
+  // runs every 10 seconds in the background to get all Async Data
+  useEffect(() => {
+    const interval = setInterval(() => {
```
Comment (hopetambala) [inline domains/DataCollection/Forms/PatientID/index.js:33]: this whole chain of logic needs to be put into a `util.js` file probably. it's a tad long. make a `utils` folder in the domain `PatientID` and export a function that handles all this logic so we can remove it from the form

separation of concerns

File: modules/async-storage/index.js [hunk 2]
```diff
@@ -14,6 +14,12 @@ const deleteData = async (storageName) => {
   await AsyncStorage.removeItem(storageName);
 };
 
+const getAllData = async () => {
```
Comment (hopetambala) [inline modules/async-storage/index.js:17]: great add

File: domains/SignIn/index.js [hunk 3]
```diff
@@ -86,6 +87,12 @@ const SignIn = ({ navigation }) => {
     I18n.locale = lang;
   };
 
+  // checks whether user is connected to internet, return true if connected, false otherwise
+  async function checkOnlineStatus() {
```
Comment (hopetambala) [inline domains/SignIn/index.js:91]: great function. 

you should create a module named `offline`  in the `modules` folder and move functions related to offline data collection like this to it. `expo-network` imports shouldn't be in any of the forms

File: domains/DataCollection/Forms/PatientID/index.js [hunk 4]
```diff
@@ -23,10 +27,52 @@ import configArray from './config';
...
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
+
   const toRoot = () => {
     navigation.navigate('Root');
   };
 
+  const generateRandomID = () => Math.random().toString(20).substr(2, 12);
```
Comment (hopetambala) [inline domains/DataCollection/Forms/PatientID/index.js:66]: we should create a `utils` folder at the root of the project and export this to form it

**Review by hopetambala — COMMENTED** [review]
Let me know when it's refactored!

**Comment by hopetambala** [comment]
#110 closes this issue @jamccomb92 

Offline work is merged to master

**Pull request closed without merging**